lib/user.rb

```diff
@@ -0,0 +1,60 @@
+require './lib/library.rb'
+require 'date'
+
+class User
+  attr_accessor :account
+
+  def initialize
+    @account = []
+    @library = Library.new
+  end
+
+  def find_available_books
+    @library.find_available_books
+  end
+
+  def search_for_book_title(search_str)
+    search_results = []
+    search = @library.data.select { |book| book[:item][:title].include? search_str}
+    search.each { |book| search_results << "#{book[:item][:title]} : #{book[:item][:author]}" }
+    return search_results
+  end
+
+  def search_for_book_author(search_str)
+    search_results = []
+    search = @library.data.select { |book| book[:item][:author].include? search_str}
+    search.each { |book| search_results << "#{book[:item][:author]} : #{book[:item][:title]}" }
+    return search_results
+  end
+
+  def checkout(title, author)
+    data = @library.data
+    transaction_success = false
+    for book in (0...data.length) do
+      if (data[book][:item][:author] == author) && (data[book][:item][:title] == title)
+        if data[book][:available] == true
```

Comment:
is it possible to extract the if statement, rather than having nested if statements?